File: eTheater.Model/Hall.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eTheater.Model
{
    public class Hall
    {
        public int HallId { get; set; }

        public string Name { get; set; }

        public int TotalSeats { get; set; }

        public int TotalRows { get; set; }

        public int NumberOfSeatsPerRow { get; set; }

        public bool IsDeleted { get; set; }
    }
}

File: UI/etheater_web/lib/models/hall.dart
import 'package:json_annotation/json_annotation.dart';

part 'hall.g.dart';

@JsonSerializable()
class Hall {
  int hallId;
  String name;
  int totalSeats;
  int totalRows;
  int numberOfSeatsPerRow;
  bool? isDeleted;

  Hall(
      {required this.hallId,
      required this.name,
      required this.totalSeats,
      required this.totalRows,
      required this.numberOfSeatsPerRow,
      this.isDeleted});

  factory Hall.fromJson(Map<String, dynamic> json) => _$HallFromJson(json);
  Map<String, dynamic> toJson() => _$HallToJson(this);
}
